homeassistant/components/sonarr/sensor.py

```diff
@@ -72,198 +61,481 @@
-        elif self.type == "wanted":
-            for show in self.data:
-                attributes[
-                    show["series"]["title"]
-                    + " S{:02d}E{:02d}".format(
-                        show["seasonNumber"], show["episodeNumber"]
-                    )
-                ] = show["airDate"]
-        elif self.type == "commands":
-            for command in self.data:
-                attributes[command["name"]] = command["state"]
-        elif self.type == "diskspace":
-            for data in self.data:
-                attributes[data["path"]] = "{:.2f}/{:.2f}{} ({:.2f}%)".format(
-                    to_unit(data["freeSpace"], self._unit),
-                    to_unit(data["totalSpace"], self._unit),
-                    self._unit,
-                    (
-                        to_unit(data["freeSpace"], self._unit)
-                        / to_unit(data["totalSpace"], self._unit)
-                        * 100
-                    ),
-                )
-        elif self.type == "series":
-            for show in self.data:
-                if "episodeFileCount" not in show or "episodeCount" not in show:
-                    attributes[show["title"]] = "N/A"
-                else:
-                    attributes[show["title"]] = "{}/{} Episodes".format(
-                        show["episodeFileCount"], show["episodeCount"]
-                    )
-        elif self.type == "status":
-            attributes = self.data
-        return attributes
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes of the entity."""
+        attrs = {}
+
+        for command in self._commands:
+            attrs[command.name] = command.state
+
+        return attrs
 
     @property
-    def icon(self):
-        """Return the icon of the sensor."""
-        return self._icon
+    def state(self) -> Union[None, str, int, float]:
+        """Return the state of the sensor."""
+        return len(self._commands)
+
+
+class SonarrDiskspaceSensor(SonarrSensor):
+    """Defines a Sonarr Disk Space sensor."""
+
+    def __init__(self, sonarr: Sonarr, entry_id: str, unique_id: str) -> None:
+        """Initialize Sonarr Disk Space sensor."""
+        self._disks = []
+        self._total_free = 0
+
+        super().__init__(
+            sonarr=sonarr,
+            entry_id=entry_id,
+            unique_id=unique_id,
+            icon="mdi:harddisk",
+            key="diskspace",
+            name=f"{sonarr.app.info.app_name} Disk Space",
+            unit_of_measurement=DATA_GIGABYTES,
+            enabled_default=False,
+        )
+
+    def _to_unit(self, value):
+        """Return a value converted to unit of measurement."""
+        return value / 1024 ** BYTE_SIZES.index(self._unit_of_measurement)
+
+    @sonarr_exception_handler
+    async def async_update(self) -> None:
+        """Update entity."""
+        app = await self.sonarr.update()
```

Comment:
There are multiple entities calling this. That's not good. You should use a Data Update Coordinator.

Reply:
would it be acceptable to do this in followup PR given size of this one?